ldk/javascript/examples/self-test-loop/src/tests/document/index.tsx

```diff
@@ -0,0 +1,112 @@
+          hiddenRows: [],
+          name: 'name',
+          rows: [{ cells: [{ value: 'value' }] }],
+        },
+      ],
+    };
+    setTimeout(() => {
+      reject(new Error('XLSX encode did not finish in the appropriate time span.'));
+    }, 5000);
+
+    try {
+      const uint8ArrayData = await document.xlsxEncode(workbook);
+      const actual = await document.xlsxDecode(uint8ArrayData);
+      const cellData = actual.worksheets[0].rows[0].cells[0].value;
+
+      if (cellData === 'value') {
+        resolve(true);
+      } else {
+        reject(
+          new Error(`XLSX function  ${document.xlsxEncode} and ${document.xlsxDecode} failed.`),
+        );
+      }
+    } catch (e) {
+      console.error(e);
+      reject(e);
+    }
+  });
+
+export const testDocumentReadPDF = (): Promise<boolean> =>
+  new Promise(async (resolve, reject) => {
+    try {
+      let request = await network.httpRequest({
+        url: 'https://github.com/open-olive/loop-development-kit/raw/main/ldk/javascript/examples/self-test-loop/static/ldk-pdf-test.pdf',
+        method: 'GET',
+      });
+      let branch = 'main';
+
+      // Above URL is used after this feature is finished and merged in
+      // Before PR is merged and branch is deleted, use the feature branch
+      // Can delete this block after merge
+      if (request.statusCode === 404) {
+        request = await network.httpRequest({
+          url: 'https://github.com/open-olive/loop-development-kit/raw/HELPS-3035-readpdf/ldk/javascript/examples/self-test-loop/static/ldk-pdf-test.pdf',
+          method: 'GET',
+        });
+        branch = 'HELPS-3035-readpdf';
+      }
+
+      const decoded = await document.readPDF(request.body);
+
+      // Sometimes the Core returns the pages out of order, so this is to sort it
+      const decodedSorted: PDFOutput = {};
+      for (let i = 1; i < Object.keys(decoded).length + 1; i += 1) {
+        decodedSorted[i] = { ...decoded[i] };
+      }
+
+      let pdfParse = '';
+
+      Object.entries(decodedSorted).forEach(([page, { Content }]) => {
+        pdfParse += `# Page ${page}\n`;
+
+        Content.forEach(({ Value, Type }) => {
+          if (Type === PDFContentType.Text) {
+            pdfParse += Value;
+          } else if (Type === PDFContentType.NewLine) {
+            pdfParse += '\n\n';
+          }
+        });
+
+        pdfParse += '\n\n';
+      });
+
+      const DocumentWhisper = ({ parsedText }: { parsedText: string }) => (
+        <oh-whisper label="PDF Aptitude - Read test" onClose={() => undefined}>
+          <oh-markdown
+            body={
+              'This test should be automatically downloading this PDF and parsing only \
+            the **text** from it to display it below'
+            }
+          />
+          <oh-markdown
+            body={`Link to PDF:
+            https://github.com/open-olive/loop-development-kit/blob/${branch}/ldk/javascript/examples/self-test-loop/static/ldk-pdf-test.pdf`}
```

Comment:
Might be nice to obscure this link into a markdown link so it's not all cray cray, unless reading the url is important!